src/commcare_cloud/commands/deploy/slack.py

```diff
@@ -0,0 +1,94 @@
+import requests
+from requests import RequestException
+
+
+class SlackException(Exception):
+    pass
+
+
+def notify_slack_deploy_start(environment, context):
+    try:
+        client = SlackClient(environment)
+    except SlackException:
+        return
+
+    client.send_deploy_start_message(context)
+
+
+def notify_slack_deploy_end(environment, context, is_success):
+    try:
+        client = SlackClient(environment)
+    except SlackException:
+        return
+
+    client.send_deploy_end_message(context, is_success)
```

Comment:
What's the rationale behind putting this outside the `try/except`?  It's not that unlikely for this integration to break with a version change or something, so this will force us to fix it, but getting an exception during the deploy is pretty painful.  That used to happen with some regularity on record_deploy_success, and there was often confusion about whether the deploy needed to be redone or whether we needed to do something to make sure the environment was in a good state.

An alternative would be to loudly report the failure, but let the script continue.

Reply:
updated to catch errors and output a warning message